packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -182,6 +187,15 @@ function reducer<T, C>(data: AsyncListState<T, C>, action: Action<T, C>): AsyncL
             items: action.type === 'loading' ? [] : data.items,
             abortController: action.abortController
           };
+        case 'update':
```

Comment:
Added `update` case handling now that `selectedKeys/setSelectedKeys` from `useAsyncList` is being used instead of a user defined `selectedKey` tracker